agreement/msgp_gen.go

```diff
@@ -1691,17 +1692,17 @@ func (z *proposal) UnmarshalMsg(bts []byte) (o []byte, err error) {
 		}
 		if zb0004 > 0 {
 			zb0004--
-			bts, err = (*z).unauthenticatedProposal.Block.BlockHeader.TxnCommitments.Sha256Commitment.UnmarshalMsg(bts)
+			bts, err = (*z).unauthenticatedProposal.Block.BlockHeader.TxnCommitments.NativeSha512_256Commitment.UnmarshalMsg(bts)
 			if err != nil {
-				err = msgp.WrapError(err, "struct-from-array", "Sha256Commitment")
+				err = msgp.WrapError(err, "struct-from-array", "NativeSha512_256Commitment")
```

Comment:
@Aharonee @algoidan wanted to makes sure this was as expected?

Reply:
I think I know what happened.
[`04814e1` (#3829)](https://github.com/algorand/go-algorand/pull/3829/commits/04814e1a929472d4e66d147d0d95bd4fa9a03d34#diff-da02af82e8255801f2338dd5681ec7e0ae05caf0e1a4c8ae4b34fce4b7c3613e)

This is the commit where I've renamed the `TxnRoot` to `TxnCommitments`, and its nested fields.
While doing that I've regenerated msgp, but after noticing the fields in `TxnCommitments` where no longer in alphabetical order, I swapped them.
```
	TxnCommitments struct {
		_struct struct{} `codec:",omitempty,omitemptyarray"`

		Sha256Commitment crypto.Digest `codec:"txn256"`
		NativeSha512_256Commitment crypto.Digest `codec:"txn"`
```
to
```
	TxnCommitments struct {
		_struct struct{} `codec:",omitempty,omitemptyarray"`

		NativeSha512_256Commitment crypto.Digest `codec:"txn"`
		Sha256Commitment crypto.Digest `codec:"txn256"`
```

This doesn't break any tests of course, and wasn't caught the the `codegen_verification` script, but when generating msgp after this change the resulting code will be different (it seems to be generated in a sequential manner) even though its logic is identical.